jmix-search/search/src/main/java/io/jmix/search/index/fileparsing/AbstractExtensionBasedFileParserResolver.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2024 Haulmont.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.jmix.search.index.fileparsing;
+
+import com.google.common.base.Strings;
+import io.jmix.core.FileRef;
+import org.apache.commons.io.FilenameUtils;
+
+import java.util.List;
+
+/**
+ * Implements the common logic for all extension based file parser resolvers.
+ */
+public abstract class AbstractExtensionBasedFileParserResolver implements FileParserResolver {
+
+    /**
+     * Returns a collection of supported extensions of the supported file type. E.g. ["xlsx", "XLSX", "docx", "DOCX"].
```

Comment:
We discussed with Gleb such option but we desided not to implement it within this issue. It will be implemented in a separate issue. https://github.com/jmix-framework/jmix/issues/3696

Reply:
The JavaDoc was corrected